src/main/java/com/microsoft/graph/models/extensions/Multipart.java

```diff
@@ -33,46 +38,132 @@ public Multipart() {
+	public String getBoundary() {
 		return boundary;
 	}
 	
+	/**
+	 * Set the multipart boundary for use in the request header
+	 * @param boundary The multipart boundary
+	 */
+	public void setBoundary(String boundary) {
+		this.boundary = boundary;
+	}
+	
+	/**
+	 * Get the contentType for use in the request header
+	 * @return the multipart Content-Type
+	 */
+	public String getContentType() {
+		return contentType;
+	}
+	
+	/**
+	 * Set the contentType for use in the request header
+	 * @param contentType The multipart Content-Type
+	 */
+	public void setContentType(String contentType) {
+		this.contentType = contentType;
+	}
+	
 	/**
 	 * Get the Content-Type header to send the multipart request
 	 * @return the multipart header option
 	 */
 	public HeaderOption header() {
-		return new HeaderOption("Content-Type", "multipart/form-data; boundary=\"" + boundary + "\"");
+		return new HeaderOption("Content-Type", contentType + "; boundary=\"" + boundary + "\"");
+	}
+	
+	private void writePartData(String partContent, byte[] byteArray) throws IOException{
+		out.write(partContent.getBytes(MULTIPART_ENCODING));
+		out.write(byteArray);
+		String returnContent = RETURN + RETURN;
+		out.write(returnContent.getBytes(MULTIPART_ENCODING));
+	}
+	
+	private String createPartHeader(Map<String, String> headers, String name, String contentType, String filename) {
+		String partContent = addBoundary();
+		if(headers != null) {
+			List<String> listContentDisposition = Arrays.asList("filename","creation-date","modification-date","read-date",
+					"size","name","voice","handling","preview-type");			
+			List<String> mainHeaders = Arrays.asList("Content-Disposition","Content-Type","charset");
+			
+			if(headers.containsKey("Content-Disposition")) {			
+				partContent += "Content-Disposition:"+headers.get("Content-Disposition")+";"; 
+				for (Map.Entry<String,String> entry : headers.entrySet()) {
+					if(listContentDisposition.contains(entry.getKey())) {
+						partContent += " " + entry.getKey() + "=\"" + entry.getValue() + "\";";
+					}
+				}
+				partContent = partContent.substring(0, partContent.length()-1);
+				partContent += RETURN;
+				
+			}
+			
+			if(headers.containsKey("Content-Type")) {
+				partContent += "Content-Type:"+headers.get("Content-Type")+";"; 
+				if(headers.containsKey("charset")) {
+					partContent += "charset=\"" + headers.get("charset") + "\";"; 
+				}
+				partContent = partContent.substring(0, partContent.length()-1);
+				partContent += RETURN;
+			}
+			
+			for(Map.Entry<String,String> entry : headers.entrySet()) {
+				if(mainHeaders.contains(entry.getKey())==false && listContentDisposition.contains(entry.getKey())==false) {
+					partContent += entry.getKey() +":"+entry.getValue() + RETURN;
+				}
+			}
+			partContent += RETURN;
+		}
+		else if(filename != null && name != null) {
+			partContent +=
+					"Content-Disposition:form-data; name=\"" + name + "\"" + "; filename=\"" + filename + "\"" + RETURN +
```

Comment:
There needs to be a space after the colon in the Content-Disposition header.

Reply:
Added space after the colon in the Content-Disposition header.